.github/workflows/terraform-destroy-environment-manual.yml

```diff
@@ -52,46 +52,63 @@ jobs:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.inputs.build_branch}}
+          repository: 'nhsconnect/national-document-repository-infrastructure'
+          fetch-depth: '0'
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ secrets.AWS_ASSUME_ROLE }}
+          role-skip-session-tagging: true
+          aws-region: ${{ vars.AWS_REGION }}
+
+      - name: View AWS Role
+        run: aws sts get-caller-identity
+
+      # Install the latest version of Terraform CLI and configure the Terraform CLI configuration file with a Terraform Cloud user API token
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: 1.5.4
+          terraform_wrapper: false
 
-    - name: Configure AWS Credentials
-      uses: aws-actions/configure-aws-credentials@v2
-      with:
-        role-to-assume: ${{ secrets.AWS_ASSUME_ROLE }}
-        role-skip-session-tagging: true
-        aws-region: ${{ vars.AWS_REGION }}
-      
-    - name: View AWS Role
-      run: aws sts get-caller-identity
+      - name: Terraform Init
+        id: init
+        run: terraform init -backend-config=${{ github.event.inputs.backend }}
+        working-directory: ./infrastructure
+        shell: bash
 
+      - name: Terraform Set Workspace
+        id: workspace
+        run: terraform workspace select ${{ github.event.inputs.sandbox_workspace}}
+        working-directory: ./infrastructure
+        shell: bash
 
-    # Install the latest version of Terraform CLI and configure the Terraform CLI configuration file with a Terraform Cloud user API token
-    - name: Setup Terraform
-      uses: hashicorp/setup-terraform@v2
-      with:
-        terraform_version: 1.5.4
-        terraform_wrapper: false
+      - name: Get CloudFront Distribution ID
+        id: cloudfront
+        run: echo "distribution_id=$(aws cloudfront list-distributions |jq -r '.[]["Items"][] | select(.Origins.Items[].Id=="${{inputs.sandbox_workspace}}-lloyd-george-store") | .Id')" >> $GITHUB_OUTPUT
 
-    - name: Terraform Init
-      id: init
-      run: terraform init -backend-config=${{ github.event.inputs.backend }}
-      working-directory: ./infrastructure
-      shell: bash
+      # Install dependencies for Python scripts
+      - name: Install Python and Dependencies
+        id: dependencies
+        run: |
+          python3 -m venv ./venv
+          ./venv/bin/pip3 install --upgrade pip
+          ./venv/bin/pip3 install boto3==1.33.11
 
-    - name: Terraform Set Workspace
-      id: workspace
-      run:  terraform workspace select ${{ github.event.inputs.sandbox_workspace}}
-      working-directory: ./infrastructure
-      shell: bash
+      # Run script to remove Lambda@Edge associations
+      - name: Remove Lambda@Edge & CloudFront associations
```

Comment:
Might be worth extracting out into it's own job (called `remove_edge_associations`) similar to the cleanup versions, just so you can put a dependency on this being successful before trying to destroy, and on line 52 update to `needs: [cleanup_versions_process, remove_edge_associations]`